Folia

ricsal1 opened the pull request:
More support for folia
 Configurable quitExitGame config so player will leave game if leaves server or not (default false)
 Configurable alwaysDay config so that it always stays day or not (default true)
 Configurable randomMaps config so that it can change map at every round, case you have mode with more then one (default false)
 Fixed problem with duplicated game counter
 Fixed problem with items pickup

Commit:
CSMC v0.9 by tontito007:
 Initial support for Folia
 Updated game engine to be prepared to have several game modes

ricsal1 committed:
CSMC v0.9.4 by tontito007:
 Cleaned code
 Optimized folia detection for newer versions
 Added protections for the initial setups in order to register correctly the spawns
 Chicken now spwan at spwans
 Cleanup Entities now work better on both latest Paper and Folia (missing support on quality guns)
 New scoreBoard system shared both for Paper and Folia

ricsal1 committed:
CSMC v0.9.5 by tontito007:
 Fixed player death at some cases
 Doors cant be open if you have bomb in hand
 Fast Board for score boards
 Better folia support

ricsal1 committed:
CSMC v0.9.7 by tontito007:
 Configurable quitExitGame config so player will leave game if leaves server or not (default false)
 Configurable alwaysDay config so that it always stays day or not (default true)
 Configurable randomMaps config so that it can change map at every round, case you have mode with more then one (default false)
 Fixed problem with duplicated game counter
 Fixed problem with items pickup

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/commands/CounterStrikeCommand.java:
```diff
@@ -163,9 +103,17 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
+//                main.saveConfig();
+//
+                if (Map == null) {
+                    player.sendMessage(ChatColor.GOLD + "Map no set, assuming " + world);
```
```suggestion
                    player.sendMessage(ChatColor.GOLD + "Map not set, assuming " + world);

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/commands/CounterStrikeCommand.java:
```diff
@@ -181,9 +129,14 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
-                plugin.LoadCOnfigs();
+                plugin.loadConfigs();
+
+                if (Map == null) {
+                    player.sendMessage(ChatColor.GOLD + "Map no set, assuming " + world);
+                    Map = world;
+                }
```
This might be moved to a external method, as it is repeated 3 times

ricsal1 replied:
for now will leave at it is, but with spelling corrected :)

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java:
```diff
@@ -28,6 +28,7 @@ public class CSPlayer {
     private Player player;
     private int money;
     private int kills;
+    private int chickenkills;
```
```suggestion
    private int chickenKills;

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java:
```diff
@@ -37,9 +38,10 @@ public class CSPlayer {
     private String opponentcolour;
```
```suggestion
    private String opponentColour;

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java:
```diff
@@ -119,15 +126,16 @@ public CSPlayer(CounterStrike plugin, Player player, String colour) {
+
         if (player.getInventory().getItemInMainHand().getType().equals(Material.IRON_AXE)) {
-            player.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 99999, Config.KNIFE_SPEED - 2));
+            CounterStrike.i.myBukkit.runTask(player,null,null, () -> player.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 99999, (int) (player.getWalkSpeed() * 0.7))));
```
I think the KNIFE_SPEED from config is no longer used - maybe remove it or include it in the speed formula

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/events/WeaponFireEvent.java:
```diff
@@ -1,47 +1,48 @@
+
+    Player shooter;
+    Weapon gun;
+    Projectile bullet;
```
```suggestion
    private Player shooter;
    private Weapon gun;
    private Projectile bullet;
```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/EntityPickupItemListener.java:
```diff
@@ -45,33 +44,41 @@ public void onPlayerPickupItem(EntityPickupItemEvent e) {
                         }
                         e.setCancelled(true);
-                        break;
+
```
I don't quite understand why to remove break here. After removing it, if `RIFLE` is matched, the code of `RIFLE`, `PISTOL` and `GRENADE` will run until break; from `GRENADE`.

ricsal1 replied:
Took some time to remember, the problem of not documenting when we find a problem...
This is because you always want to allow to pickup a grenade, but you don't want to for a rifle or pistol if you already have one.
Having a break will not cancel the event and the player will get the item picked + the item set just before on the right slot.
Basically is is to force items on the right slot.

barpec12 replied:
I think you didn't get my point. If you have such switch:
```
switch(status) {
  case ALIVE {
   System.out.println("I'm alive!");
  }
  
  case DEAD {
   System.out.println("I'm dead!");
  }
}
```

When status is DEAD, it will print
`I'm dead!`

BUT if status will be ALIVE, it will print both messages:
`I'm alive!
I'm dead!`.

We need `break` after each case, because instead every case, that is below is called. (see for example https://www.reddit.com/r/learnjava/comments/susdi2/why_does_switch_without_break_execute_all_cases/).

barpec12 replied:
So in this case, if Weapon will be RIFLE and player will not have a Pistol, a RIFLE will be first set to the slot 0 and then slot 1

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/EntityPickupItemListener.java:
```diff
@@ -45,33 +44,41 @@ public void onPlayerPickupItem(EntityPickupItemEvent e) {
+                            p.getInventory().setItem(1, item);
                         }
                         e.setCancelled(true);
-                        break;
```
Like the previous comment

ricsal1 replied:
same as before, noo need to have break if you are not running more code

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerChatControlListener.java:
```diff
@@ -0,0 +1,41 @@
+            return;
+        }
+
+        //dead players don't talk
```
<3

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerChatControlListener.java:
```diff
@@ -0,0 +1,41 @@
+        //dead players don't talk
+        if (player.getGameMode() == GameMode.SPECTATOR) {
+        //  chat.callEvent();
+            chat.setMessage("");
```
Maybe instead of setting the message to `""`, cancel the event?
```suggestion
            chat.setCancelled(true);
```

ricsal1 replied:
On a version I have tested this resulted on a infinit loop (for whatever stange reason) resulting in crash, will retry with later paper versions

ricsal1 replied:
works well on paper 1.21. so now canceling the event

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerDeathListener.java:
```diff
@@ -39,9 +41,21 @@ public void playerDeathEvent(PlayerDeathEvent event) {
-        CSPlayer csplayerVictim = CounterStrike.i.getCSPlayer(victim,false, null);
+        CSPlayer csplayerVictim = plugin.getCSPlayer(victim, false, null);
 
+        //se morreu sozinho
```
```suggestion
        //when player died alone

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerInteractListener.java:
```diff
@@ -21,16 +22,19 @@
 
 public class PlayerInteractListener implements Listener {
 
+    CounterStrike plugin;
```
```suggestion
    private CounterStrike plugin;

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java:
```diff
@@ -33,36 +34,52 @@ public PlayerJoinListener() {
-        if (plugin.ResourseHash.get(player.getName()) == null && event.getStatus().equals(SUCCESSFULLY_LOADED)) {
-            plugin.ResourseHash.put(player.getName(), true);
+        if (!event.getStatus().equals(ACCEPTED) && !event.getStatus().equals(SUCCESSFULLY_LOADED)) {
+            Utils.debug("Estado de carregamento " + event.getStatus());
```
```suggestion
            Utils.debug("Loading state: " + event.getStatus());

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java:
```diff
@@ -75,57 +92,65 @@ public void playerJoinEvent(PlayerJoinEvent event) {
 
-        System.out.println("#### Player " + player.getName() + " entered ");
+        if (!plugin.getPlayerUpdater().playersWithScoreboard.contains(player.getUniqueId())) {
+            Utils.debug("#### Player " + player.getName() + " entered to lobby");
```
```suggestion
            Utils.debug("#### Player " + player.getName() + " entered the lobby");

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java:
```diff
@@ -75,57 +92,65 @@ public void playerJoinEvent(PlayerJoinEvent event) {
+                player.setGameMode(GameMode.SURVIVAL);
+                player.getInventory().clear();
+                player.getInventory().setArmorContents(null);
+                player.setFoodLevel(8); //era 6
```
```suggestion
                player.setFoodLevel(8); //was 6

```

or even remove this comment

```suggestion
                player.setFoodLevel(8);

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java:
```diff
@@ -75,57 +92,65 @@ public void playerJoinEvent(PlayerJoinEvent event) {
+
+            plugin.returnPlayertoGame(csplayer);
+
+            PacketUtils.sendTitleAndSubtitle(player, ChatColor.YELLOW + "Get ready", ChatColor.RED + "You will resume plaing in next round!", 1, 8, 1);
```
```suggestion
            PacketUtils.sendTitleAndSubtitle(player, ChatColor.YELLOW + "Get ready", ChatColor.RED + "You will resume playing next round!", 1, 8, 1);

```

ricsal1 replied:
some typing errors make me feel kinda uncomfortable :)

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java:
```diff
@@ -134,15 +159,14 @@ public void playerChangedWorldEvent(PlayerChangedWorldEvent event) {
         }
 
-        System.out.println("#### Player " + player.getName() + " changes world from " + event.getFrom().getName() + " to " + mundo);
+        Utils.debug("#### Player " + player.getName() + " changes world from " + event.getFrom().getName() + " to " + mundo);
```
```suggestion
        Utils.debug("#### Player " + player.getName() + " changed world from " + event.getFrom().getName() + " to " + mundo);

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/Bomb.java:
```diff
@@ -1,72 +1,91 @@
+    private static boolean cleaning = false;
+    Object global;
+    Object bombTask;
+    Object defuseTask;
```
access modifier is missing, should this be `protected`?
And also look at the comment in MyBukkit regarding `Object` type
```suggestion
    private Object global;
    private Object bombTask;
    private Object defuseTask;

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/Bomb.java:
```diff
@@ -1,72 +1,91 @@
+
+        CounterStrike.i.gameState = GameState.PLANTED;
+
+        try {
+//            this.hologram = HologramsAPI.createHologram(CounterStrike.i, location.clone().add(0.5, 2.25, 0.5));
+//            hologram.appendTextLine(ChatColor.RED + "Bomb");
+//            hologram.appendTextLine(ChatColor.YELLOW + "Exploding in " + countdown + " seconds.");
+//            hologram.appendTextLine(ChatColor.GRAY + "(Right click to defuse)");
+        } catch (Exception e) {
+        }
```
```suggestion
```

ricsal1 replied:
What is the difference here?

barpec12 replied:
Removing this part of code

ricsal1 replied:
Holographic Displays has been discontinued but I am looking for options, like decentholograms, so keeping the code commented so later on I can look for replace the calls

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/Bomb.java:
```diff
@@ -90,85 +114,86 @@ public void defuse(CSPlayer csplayer) {
-                }
+        defuseTask = CounterStrike.i.myBukkit.runTaskTimer(null, location, null, () -> {
+            if (location == null || hologram == null) {
+                System.out.println("#################  contingency bomb cleaning");
```
Maybe use Utils to log this

ricsal1 replied:
Na, just remove that debug :)

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/utils/MyBukkit.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+
+    public Object runTask(Player player, Location local, Entity entity, Runnable myrun) {
```
I think all operations on this Object are done in this class, so instead of returning object, you could return a custom object, that would be sth like

```
class CSMCTask {
  BukkitTask taskBukkit = null;
  ScheduledTask scheduledTask = null;
}
```

and depending on the bukkit engine used, access one field or another. You would not need to cast Object to tasks, return Object to another classes, which does not look great. Returning typed object looks better

barpec12 replied:
On the second thought, having in this class ScheduledTask might not be possible, as it might throw an error on spigot, which will not know, what is this type. In that case it might be 
```
class CSMCTask {
  BukkitTask taskBukkit = null;
  Object scheduledTask = null;
}
```
This is not that great, but in other classes you would use type CSMCTask instead of Object

ricsal1 replied:
Been thinking from time to time about it, for now, considering my limitations, this seems the best option

Later on it could implement something like a dictionary and manage internally these objects from a general key

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/Bomb.java:
```diff
@@ -186,7 +211,7 @@ public DefuseChecker(CSPlayer defuser) {
         @EventHandler
         private void weaponFire(WeaponFireEvent event) {
-            System.out.println("defusing weapon fire event");
+            System.out.println("############## defusing weapon fire event");
```
This can be also logged using Utils

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/GameCounter.java:
```diff
@@ -29,33 +27,48 @@ public GameCounter(CounterStrike plugin) {
     }
 
-    @Override
+    //CouterStrike->GameCounter (aguarda que jogadores digam que querem jogar)->GameStarter (espera pelo minimo de jogadores e passa a starting)->GameTimer
+    //  GameState.LOBBY    - qd jogadores entram, depois de terminar qd jogadores ficam em espera ou qd nao tem jogadores a jogar
+    //  GameState.WAITING  - qd aguarda que cheguem mais jogadores em GameCounter
+    //  GameState.STARTING - qd esta memso a começar, qd arrnaca GameStarter
+
```
Those comments could be translated to English

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/PlayerUpdater.java:
```diff
@@ -62,82 +68,41 @@ public void run() {
 
-        Scoreboard board = Bukkit.getScoreboardManager().getNewScoreboard();
+        FastBoard board = new FastBoard(player);
+        board.updateTitle(ChatColor.BOLD + "----Miner Strike v" + CounterStrike.i.getDescription().getVersion() + "----");
```
This probably should be configurable

ricsal1 replied:
Scoreboard are no longer supported in Folia, so just changed everything to a way of doing the same in all types of servers

barpec12 replied:
I mean the name of the title `----Miner Strike v`

ricsal1 replied:
That should be our "brand" name, we can do a little brain storm a get a more definitive one.
If we make it configurable we will start seeing sings like DJMaxKiller Plugin version xpto :P

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/runnables/Reloader.java:
```diff
@@ -1,40 +1,45 @@
+        if (player.getLevel() >= 1) {
+            item.setAmount(gun.getAmmunition());
+
+            System.out.println(gun.getAmmunition() + " reload gun " + gun.getMagazineCapacity());
```
debug using Utils

ricsal1 replied:
no longer needed debug, thanks :)

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/shop/Shop.java:
```diff
@@ -120,12 +123,19 @@ public void purchaseShopItem(Player player, Weapon gun) {
+                    try {
+                        helmetMeta.setColor(mycolor);
+                    } catch (Exception e) {
+                        System.out.println(e.getMessage() + " ########### Erro em Item111 " + gun.loadItem());
```
Use utils and translate to English

ricsal1 replied:
better remove the debug :)

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/shop/Shop.java:
```diff
@@ -120,12 +123,19 @@ public void purchaseShopItem(Player player, Weapon gun) {
+                    helm.setItemMeta(helmetMeta);
+                    player.getInventory().setHelmet(helm);
+                } catch (Exception e) {
+                    System.out.println(e.getMessage() + " ########### Erro em Item " + gun.loadItem());
```
Use utils and translate to English

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/shop/Shop.java:
```diff
@@ -150,6 +160,9 @@ public void purchaseShopItem(Player player, Weapon gun) {
         }
 
         if (CounterStrike.i.usingQualityArmory() && type != WeaponType.GRENADE && type != WeaponType.HELMET && type != WeaponType.ARMOUR) {
+            me.zombie_striker.qg.guns.Gun gun1 = me.zombie_striker.qg.api.QualityArmory.getGunByName(gun.getName());
```
We have now hard dependency on QualityArmory, so absolute reference is not needed anymore, I think
```suggestion
            Gun gun1 = QualityArmory.getGunByName(gun.getName());

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/shop/Shop.java:
```diff
@@ -150,6 +160,9 @@ public void purchaseShopItem(Player player, Weapon gun) {
+            me.zombie_striker.qg.guns.Gun gun1 = me.zombie_striker.qg.api.QualityArmory.getGunByName(gun.getName());
+            Gun.updateAmmo(gun1, player.getInventory().getItem(0), gun.getMagazineCapacity());
+
             ItemStack ammo = me.zombie_striker.qg.api.QualityArmory.getGunByName(gun.getName()).getAmmoType().getItemStack().clone();
```
```suggestion
            ItemStack ammo = QualityArmory.getGunByName(gun.getName()).getAmmoType().getItemStack().clone();

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/utils/PacketUtils.java:
```diff
@@ -2,107 +2,109 @@
+    }
+
+
+    private static Boolean InLobbyLocation(Player p) {
```
function name should start with lowercase letter
```suggestion
    private static Boolean isInLobbyLocation(Player p) {

```

barpec12 reviewed (changes requested):
Hi, I reviewed your changes. I am amazed with your improvements and changes. Well done! I have some suggestions, but most of them are not that important. You might just ignore them. I think the most important one is about the `switch` with `RIFLE` and so on. This seem like a bug.

ricsal1 committed:
Update CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java

Co-authored-by: barpec12 <kontakt@barpec12.pl>

ricsal1 committed:
CSMC v0.9.8 by tontito007:
 Fixed problems reported from PR
 Renamed some variables
 Deleted some commented code

ricsal1 committed:
CSMC v0.9.8 by tontito007:
 Fixed problems reported from PR
 Renamed some variables
 Deleted some commented code
 Updated dependencies in plugin.yml
 Reorganized PlayerInteractListener event considering the actions
 Code cleanup

ricsal1 committed:
Merge remote-tracking branch 'origin/Folia' into Folia

# Conflicts:
#	CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java

ricsal1 committed:
CSMC v0.9.8 by tontito007:
 Fixed problems reported from PR
 Renamed some variables
 Deleted some commented code
 Updated dependencies in plugin.yml
 Reorganized PlayerInteractListener event considering the actions
 Code cleanup

ricsal1 committed:
CSMC v0.9.8 by tontito007:
 Fixed problems reported from PR
 Renamed some variables
 Deleted some commented code
 Updated dependencies in plugin.yml
 Reorganized PlayerInteractListener event considering the actions
 Code cleanup
